src/sentry/snuba/metrics/datasource.py

```diff
@@ -189,7 +189,8 @@ def _get_metrics_filter_ids(projects: Sequence[Project], metric_names: Sequence[
                     derived_metric_obj.naively_generate_singular_entity_constituents()
                 )
                 metric_names_deque.extend(single_entity_constituents)
-    if None in metric_ids:
+
+    if None in metric_ids or -1 in metric_ids:
```

Comment:
`-1` is not a valid metric id or tag id to send to ClickHouse

Reply:
-1 is returned by the `resolve_weak` helper, see https://github.com/getsentry/sentry/blob/e063ea75be554e5e4a237d619645d7924a304074/src/sentry/sentry_metrics/utils.py#L6-L7

@ahmedetefy should `generate_metric_ids` call `indexer.resolve` instead of `resolve_weak`, so that we are consistent with line 182?

Reply:
Yes I agree with that
@MeredithAnya I will put a PR after you merge yours to incorporate Joris' suggestion